### PR TITLE
#167749950 User email verification 

### PR DESCRIPTION
**Pull request by ibraheem-adeyemo**
#### What does this PR do?
- It verifies users email

#### Description of task to be completed
- Have verifyEmail function that sends mail to a user
- Have api/v1/verifyEmail/:token route that user reach when clicking the link 
- It updates isVerified column to true

#### How should this be manually tested?
After cloning the repo cd into it and run npm install and npm test
- using the postman test 
- make signup request with the firstname, lastname, email, password.
- check the mail and click the link.

Key: content-type value: application/json

Key: Authorization value: JWT TOKEN

#### What are the relevant pivotal tracker stories
[#167749950](https://www.pivotaltracker.com/story/show/167749950)

### CODE REVIEW

**Review by Naimaoye — COMMENTED** [review]
You did a good job on this. I dont think I have any request for now. Well done!

File: .travis.yml [hunk 1]
```diff
@@ -9,6 +9,11 @@ before_script:
 node_js:
 - 'stable'
 
+env:
```
Comment (hustlaviola) [inline .travis.yml]: You can remove the environment variables. It has been added on Travis

File: src/middlewares/userValidations.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ const validateUser = path => (req, res, next) => {
   if (_.has(Schemas, path)) {
     const schema = _.get(Schemas, path);
     const response = Joi.validate(user, schema, { abortEarly: false });
-    const errors = Helper.buildErrorResponse(response);
+    const errors = Helper.handleValidation(response);
```
Comment (hustlaviola) [inline src/middlewares/userValidations.js]: You should revert the changes in this file

File: src/middlewares/userValidations.js [hunk 3]
```diff
@@ -31,8 +31,7 @@ const validateUser = path => (req, res, next) => {
  * @returns {object} JSON response
  */
 const emailExists = (req, res, next) => {
-  let { email } = req.body;
-  email = email.trim().toLowerCase();
+  const { email } = req.body;
```
Comment (hustlaviola) [inline src/middlewares/userValidations.js]: `email` has been trimmed and changed to lowercase.

File: src/routes/index.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import express from 'express';
```
Comment (hustlaviola) [inline src/routes/index.js]: This file has been deleted already. You should remove it

File: src/utils/Helper.js [hunk 5]
```diff
@@ -21,14 +54,14 @@ export default class Helper {
...
-   * @description Set error response
+   * @method handleValidation
+   * @description Handle validation errors
    * @static
    * @param {object} data
    * @returns {object} JSON response
    * @memberof Helper
    */
-  static buildErrorResponse(data) {
+  static handleValidation(data) {
```
Comment (hustlaviola) [inline src/utils/Helper.js]: This method's name does not need to change. Please revert it.

**Review by hustlaviola — CHANGES_REQUESTED** [review]
Awesome @Ibraheem-adeyemo. However, some changes were not necessary. It probably occurred when you were rebasing, maybe you accepted `incoming change` where you should have accepted the `current change`. Please check the inline comments to find the errored areas. You also need to write some test cases for this endpoint and document with Swagger. Good job

**Review by alatos2 — APPROVED** [review]

File: src/utils/Helper.js (hunk 5 above)
Comment (ibraheem-adeyemo) [reply src/utils/Helper.js]: Changes implemented

File: src/routes/index.js (hunk 4 above)
Comment (ibraheem-adeyemo) [reply src/routes/index.js]: All right 

Removed

File: src/middlewares/userValidations.js (hunk 3 above)
Comment (ibraheem-adeyemo) [reply src/middlewares/userValidations.js]: Changes implemented

File: src/middlewares/userValidations.js (hunk 2 above)
Comment (ibraheem-adeyemo) [reply src/middlewares/userValidations.js]: Done!

File: .travis.yml (hunk 1 above)
Comment (ibraheem-adeyemo) [reply .travis.yml]: Implemented

File: src/middlewares/userValidations.js (hunk 3 above)
Comment (hustlaviola) [inline src/middlewares/userValidations.js]: This need to be trimmed and converted to lower case.
```suggestion
  let { email } = req.body;
  email = email.trim().toLowerCase();
```

File: .eslintrc.json [hunk 6]
```diff
@@ -41,5 +41,4 @@
         }
     }]
     }
-}
-
+ }
```
Comment (hustlaviola) [inline .eslintrc.json]: You can add a newline here

**Review by ldonjibson — APPROVED** [review]
Good @Ibraheem-adeyemo

File: src/database/models/user.js [hunk 7]
```diff
@@ -1,22 +1,11 @@
...
-      allowNull: false,
-      unique: true
-    },
-    password: {
-      type: DataTypes.STRING,
-      allowNull: false
-    },
+    firstName: { type: DataTypes.STRING, allowNull: false },
+    lastName: { type: DataTypes.STRING, allowNull: false },
+    email: { type: DataTypes.STRING, allowNull: false },
```
Comment (hustlaviola) [inline src/database/models/user.js]:
```suggestion
    email: { type: DataTypes.STRING, allowNull: false, unique: true },
```

File: src/database/migrations/20190901204457-create-user.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      type: Sequelize.STRING,
+      allowNull: false
+    },
+    lastName: {
+      type: Sequelize.STRING,
+      allowNull: false
+    },
+    email: {
+      type: Sequelize.STRING,
+      allowNull: false
```
Comment (hustlaviola) [inline src/database/migrations/20190901204457-create-user.js]:
```suggestion
      allowNull: false,
      unique: true
```

**Review by hustlaviola — APPROVED** [review]

**Review by daviddamilola — APPROVED** [review]
LGTM
